pkg/commands/git_commands/commit_loader.go

```diff
@@ -223,7 +223,8 @@ func (self *CommitLoader) getHydratedRebasingCommits(rebaseMode enums.RebaseMode
 	for _, rebasingCommit := range commits {
 		if rebasingCommit.Sha == "" {
 			hydratedCommits = append(hydratedCommits, rebasingCommit)
-		} else if commit := fullCommits[rebasingCommit.Sha]; commit != nil {
+		} else if commit, found := lo.Find(fullCommits,
+			func(c *models.Commit) bool { return strings.HasPrefix(c.Sha, rebasingCommit.Sha) }); found {
```

Comment:
I didn't do any measurements, but I doubt that the performance impact of the linear search is significant. If we are concerned about this, we could do the slice/linear search version only if the git version is older than 2.25.2, and use the map otherwise. But this would make the code more complex, so I wouldn't be in favor of doing that.

Reply:
Also, one wonders why the last `if` doesn't have an `else` where we return an error, as this should never happen. I tried, but the rest of the code doesn't behave very well in that case: it doesn't look like we entered rebase mode (even though we did, if you check with "git status" on the command line), and it looks like any current modified files were wiped, which is pretty scary (they are auto-stashed, but you don't see it, so you have to do a `git rebase --continue` on the command line to get them back).

Fixing all this seemed out of scope for this bugfix, so I didn't change this.

Reply:
What are your thoughts on instead getting the hash length of the first rebasing commit and using that when building the map? That would be efficient regardless of which version we're on

Reply:
That wouldn't work in all cases. If I'm not mistaken, the length of abbreviated hashes is not fixed; some of them might be one character longer if they would otherwise be ambiguous.